django/db/models/indexes.py

```diff
@@ -35,8 +36,9 @@ def __init__(self, *, fields=(), name=None, db_tablespace=None, opclasses=(), co
         self.name = name or ''
         if self.name:
             errors = self.check_name()
-            if len(self.name) > self.max_name_length:
-                errors.append('Index names cannot be longer than %s characters.' % self.max_name_length)
+            if not any(k for k in ("%(app_label)s", "%(class)s") if k in self.name):
```

Comment:
```suggestion
            if not any(k for k in ("%(app_label)s", "%(class)s") if k in self.name) and len(self.name) > self.max_name_length:
```

Reply:
this was about `flake8`.  but I changed like this:
`if (not any(k for k in ("%(app_label)s", "%(class)s") if k in self.name) and`
`    len(self.name) > self.max_name_length):`